src/formElementPlugins/ApplicationViewWrapper.tsx

```diff
@@ -151,12 +150,13 @@ const ApplicationViewWrapper: React.FC<ApplicationViewWrapperProps> = (props) =>
 
   const PluginComponent = (
     <ApplicationView
+      key={JSON.stringify(evaluatedParameters)} // Forces re-render after dynamic parameter update
```

Comment:
is this really required ?
I thought parameter change would cause re-render

Reply:
It should, but for some reason it doesn't for the checkbox with dynamic options, as per my comment in the original post. Would really like to know why. Don't suppose you have any ideas? I'll have a closer look at it.

Reply:
Okay, I understand why now. But we can't use the key, that's a clunky and inefficient way to force a new instance of every element component for every change.

The problem is that the Checkbox plugin is using "derived state", in that it has a "useState" value (the checkboxes array) that is built from incoming parameters. But when the parameters change, the state doesn't get updated. I should be able to come up with a solution for this.

In the meantime, have removed the key 1e838555

Reply:
Okay, got it sorted for Checkboxes now. Made a small `useEffect` to watch for changes in `parameters.checkboxes` and update local state, but dont' actually save the state if it's still "Loading..." 1e838555